crates/project/src/lsp_command.rs

```diff
@@ -3178,3 +3183,477 @@ impl LspCommand for LinkedEditingRange {
         BufferId::new(message.buffer_id)
     }
 }
+
+impl GetDocumentDiagnostics {
+    pub fn deserialize_lsp_diagnostic(diagnostic: proto::LspDiagnostic) -> Result<lsp::Diagnostic> {
+        let start = diagnostic
+            .start
+            .ok_or_else(|| anyhow!("invalid start range"))?;
```

Comment:
```suggestion
            .context("invalid start range")?;
```

I know that there are a lot of places around doing old version, but just in case you want to change this to something more idiomatic here and below.